callouts/python/extproc/tests/redirect_test.py

```diff
@@ -0,0 +1,226 @@
+# Copyright 2024 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import print_function
+
+import datetime
+from http.server import BaseHTTPRequestHandler
+from http.server import HTTPServer
+import threading
+import time
+import urllib.request
+
+import grpc
+from grpc import ServicerContext
+import pytest
+from extproc.service import callout_server
+from envoy.config.core.v3.base_pb2 import HeaderValueOption
+from envoy.config.core.v3.base_pb2 import HeaderValue
+from envoy.service.ext_proc.v3.external_processor_pb2 import HeadersResponse
+from envoy.service.ext_proc.v3.external_processor_pb2 import BodyResponse
+from envoy.service.ext_proc.v3.external_processor_pb2 import ProcessingResponse
+from envoy.service.ext_proc.v3.external_processor_pb2 import ProcessingRequest
+from envoy.service.ext_proc.v3.external_processor_pb2 import HttpHeaders
+from envoy.service.ext_proc.v3.external_processor_pb2 import HttpBody
+from envoy.service.ext_proc.v3.external_processor_pb2_grpc import ExternalProcessorStub
+
+# Global server variable.
+server: callout_server.CalloutServer | None = None
+
+
+class CalloutServerTest(callout_server.CalloutServer):
+
+  def on_request_headers(
+      self, headers: HttpHeaders, context: ServicerContext
+  ) -> HeadersResponse:
+    """Custom processor on request headers."""
+    return callout_server.add_header_mutation(
+        add=[('host', 'service-extensions.com'), (':path', '/extensions')],
```

Comment:
dont you have to set the immediate response field for redirect?

Reply:
I was able to redirect without it, using just like this, I called /callouts for example and it redirected to the /extensions

Reply:
sure that is for the same host. But in that case, it is not a redirect. Its a rewrite. The browser's address bar does not change. A redirect requires sending HTTP 301

Reply:
Hmmm got it, I'll address the changes then